server/core_authenticate.go

```diff
@@ -477,6 +479,18 @@ func AuthenticateGoogle(ctx context.Context, logger *zap.Logger, db *sql.DB, cli
 			return "", "", false, status.Error(codes.Unauthenticated, "Error finding or creating user account.")
 		}
 
+		// Check if display name / avatar received from Google different from those in the DB
+		if dbDisplayName.String != googleProfile.Name || dbAvatarUrl.String != googleProfile.Picture {
+			logger.Info("About to check Google Id's details now. Found differences")
+			// At least one difference found, update the DB to reflect changed values
+			if _, err = db.ExecContext(ctx, "UPDATE users SET display_name = $1, avatar_url = $2, update_time = now() WHERE google_id = $3", googleProfile.Name, googleProfile.Picture, googleProfile.Sub); err != nil {
+				logger.Error("Error in updating google profile details", zap.Error(err), zap.String("googleId", googleProfile.Sub), zap.String("displayName", googleProfile.Name), zap.String("avatarUrl", googleProfile.Picture))
+				// Do no return error. Failure to update the display_name / avatar_url should not
+				// interrupt the execution. Ignore the result and the error, just log the error.
+				// return "", "", false, status.Error(codes.Unknown, "Error finding or creating user account.")
```

Comment:
We can remove this line:

```
// return "", "", false, status.Error(codes.Unknown, "Error finding or creating user account.")
```

Your text above is clear :)

Reply:
Removed in the next commit.